src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -856,12 +856,13 @@ class IMain(val settings: Settings, parentClassLoaderOverride: Option[ClassLoade
              |  ${if (resValSym != NoSymbol) s"lazy val ${lineRep.resultName} = ${originalPath(resValSym)}" else ""}
              |  lazy val ${lineRep.printName}: _root_.java.lang.String = $executionWrapper {
              |    $fullAccessPath
-             |    ( "" """.stripMargin) // the result extraction code will emit code to append strings to this initial ""
+             |    val sb = new _root_.scala.StringBuilder; """.stripMargin) // the result extraction code will emit code to append strings to this string builder
 
-          contributors map (_.resultExtractionCode(this)) foreach code.println
+             contributors.map(x => s"""sb.append("" ${x.resultExtractionCode(this)});""") foreach code.println
```

Comment:
is there a reason to generate `"" + rex` rather than just calling `sb.append(rex)`?

Reply:
Too lazy to fix the weirdness.